Subject: Ip: Fix ip -o link
Origin: 1_30_0-110-gbb3a9531b <https://git.busybox.net/busybox/commit/?id=1_30_0-110-gbb3a9531b>
Upstream-Author: Stefan Sørensen <stefan.sorensen@spectralink.com>
Date: Sat Mar 30 18:24:46 2019 +0100
Bug-Debian: https://bugs.debian.org/924374

    Commit db169f253854db572c0c2b7e3d74ebbe6afdb97f breaks the "ip -o link"
    command, no output is displayed.. Fix by only excluding the link info if
    in oneline mode and if the address family is not AF_PACKET.

    function                                             old     new   delta
    ipaddr_list_or_flush                                1232    1202     -30

    Signed-off-by: Stefan Sørensen <stefan.sorensen@spectralink.com>
    Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>

--- a/networking/libiproute/ipaddress.c
+++ b/networking/libiproute/ipaddress.c
@@ -570,8 +570,8 @@
 	}
 
 	for (l = linfo; l; l = l->next) {
-		if (no_link
-		 || (oneline || print_linkinfo(&l->h) == 0)
+		if ((oneline && G_filter.family != AF_PACKET)
+		 || (print_linkinfo(&l->h) == 0)
 		/* ^^^^^^^^^ "ip -oneline a" does not print link info */
 		) {
 			struct ifinfomsg *ifi = NLMSG_DATA(&l->h);
